data-prepper-plugins/newline-codecs/src/main/java/org/opensearch/dataprepper/plugins/codec/newline/NewlineDelimitedOutputConfig.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.codec.newline;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.List;
+
+/**
+ * Configuration class for the newline delimited codec.
+ */
+public class NewlineDelimitedOutputConfig {
+
+    @JsonProperty("header_destination")
+    private String headerDestination;
```

Comment:
Same like NewlineDelimitedInputConfig we need to identify header so that it can be added first in stream

Reply:
As in Input codec we are taking headerDestination from config , and searching for header record and write that first in eventData and then actual message , same here we are not aware where is the header and whats the header string so we are taking from config to write header first in stream and then message.

<img width="323" alt="image" src="https://github.com/opensearch-project/data-prepper/assets/133762828/ec187836-22ca-4ae2-aab6-1a4dbe658b2a">

Reply:
Do you have an example of what the output file is supposed to look like? It looks like you are writing the header per event which seems wrong to me but maybe that's the correct implementation.

Reply:
Here is the sample file screenshot, I am unable to upload NDJSON file here.

<img width="828" alt="image" src="https://github.com/opensearch-project/data-prepper/assets/133762828/cb127e81-6753-422b-9042-3f45adafab2e">

Reply:
Where is the header entry?

Reply:
In Input codec we are identifying the eader record first and putting it as header destination and remaining records as a message .

<img width="714" alt="image" src="https://github.com/opensearch-project/data-prepper/assets/133762828/214dfa58-576b-452e-9e70-39719648b112">


 same we are doing reverse of input codec , we are first fetching the header if available and then writing remaining records , so here first record considered as header record.

Reply:
So each input file will have a header? It's entirely possible that multiple input files can map to a single output file. In which case there would be multiple headers in the file. I don't understand why this header is necessary and if so why you are checking every event for it. There isn't even a header in the example file showed, so that doesn't explain the necessity for this logic.

Reply:
its not necessary that each input file shall have header, Yes Its possible that multiple input files can map to a single output file. In that case header record will be written first and then remaining records,
Note : Header doesn't contain any schema information. it just data record only , which needs to be written first and then remaining record. header record is the data record only .

Reply:
[asuresh8](https://github.com/asuresh8)  , now we will be writing in the output stream  without looking for the header destination field.